modules/distributor/receiver/shim.go

```diff
@@ -96,6 +103,21 @@ func New(receiverCfg map[string]interface{}, pusher BatchPusher, middleware Midd
 		return nil, err
 	}
 
+	for recv := range receiverCfg {
+		switch recv {
+		case "otlp":
+			receiverOtlpStats.Set(1)
+		case "jaeger":
+			receiverJaegerstats.Set(1)
+		case "zipkin":
+			receiverZipkinStats.Set(1)
+		case "opencensus":
+			receiverOpencensusStats.Set(1)
+		case "kafka":
+			receiverKafkaStats.Set(1)
```

Comment:
> is there a way to associate a string label with this "stat"? so instead of a bunch of individual stats we could says receiverStats.Type(<name>).Set(1) or something?

Discussed offline a bit, and the idea here is to centralize the stats more concretely in the usagestats package. Instead of calling something like `usagestats.NewInt("feature_enabled_search")`, `int.Set(1)` throughout the code base, this could be `usagestats.SetFeatureEnabledSearch(1)`.  The `SetFeatureEnabledSearch` method would call NewInt/Set.

Thoughts?

Reply:
The current design of the `usagesats` package is such that it doesn't care at all about what stats are being set throughout the code base.  To create some helper methods in the package _might_ help readability on the implementation side, but I'm thinking the separation of concerns here is somewhat nice.  If the variables names with a `Set()` method are ugly to look at, we could also make package local fucntions like `setFeatureEnabledSearch()` that would make the necessary calls in. 

With a package variable like `featureEnabledSearch`, calling `featureEnabledSearch.Set(1)` feels almost as readable as `usagestats.SetFeatureEnabledSearch()`, the difference being that now the usagestats needs modification for each stat we want to include.  Is there another advantage of moving the stats to the usagestats package here I'm not thinking of?

Reply:
I'm fine with either way. I like @mdisibio's suggestion to consolidate the stats b/c it will help users of the software quickly see what we are reporting. I don't consider this a blocker.

Reply:
Fair enough.  I see that it also might prevent sharing the package in the future, but I don't know how much to hold on to that idea.

Reply:
I think we could move all the variables to `modules.go` if we wanted one place to look.  We mostly need access to the config, which could be done in a helper on the Server.  Hows that sound?  I think that might smooth out a dependency loop for the backend creation also when trying to implement `backend.New()` for shared use.